src/main/java/com/faforever/client/game/CreateGameController.java

```diff
@@ -375,20 +379,37 @@ public void onRandomMapButtonClicked() {
     mapListView.scrollTo(mapIndex);
   }
 
-  public void onGenerateMapButtonClicked() {
+  public void onGenerateMapButtonClicked(){
+    onGenerateMap();
+  }
+
+  private void onGenerateMap() {
     try {
-      mapGeneratorService.generateMap().thenAccept(mapName -> {
-        Platform.runLater(() -> {
-          initMapSelection();
-          mapListView.getItems().stream()
-              .filter(mapBean -> mapBean.getFolderName().equalsIgnoreCase(mapName))
-              .findAny().ifPresent(mapBean -> {
-            mapListView.getSelectionModel().select(mapBean);
-            mapListView.scrollTo(mapBean);
-            setSelectedMap(mapBean);
+      String generatorVersion = mapGeneratorService.queryMaxSupportedVersion();
+      String majorVersion = generatorVersion.split("\\.")[0];
+      if (majorVersion.equals("1")) {
```

Comment:
Doesnt this imply that this code needs to be adjusted whenever the major version changes? Why not check for 0 and then do the old way, so we dont need to adjust this on version 2,3 etc. ?

Reply:
Makes sense done